InvenTree/InvenTree/tasks.py

```diff
@@ -306,4 +307,35 @@ def send_email(subject, body, recipients, from_email=None):
         from_email,
         recipients,
         fail_silently=False,
+        html_message=html_message
     )
+
+
+def notify_low_stock(stock_item):
+    """
+    Notify users who have starred a part when its stock quantity falls below the minimum threshold
+    """
+
+    from allauth.account.models import EmailAddress
+    starred_users = EmailAddress.objects.filter(user__starred_parts__part=stock_item.part)
+
+    if len(starred_users) > 0:
+        logger.info(f"Notify users regarding low stock of {stock_item.part.name}")
+        body = f'Hi, {stock_item.part.name} is low on stock. Kindly do the needful.'
+        context = {
+            'part_name': stock_item.part.name,
+            # Part url can be used to open the page of part in application from the email.
+            # It can be facilitated when the application base url is accessible programmatically.
+            # 'part_url': f'{application_base_url}/part/{stock_item.part.id}',
+
+            'message': body,
+
+            # quantity is in decimal field datatype. Since the same datatype is used in models,
+            # it is not converted to number/integer,
+            'part_quantity': stock_item.quantity,
+            'minimum_quantity': stock_item.part.minimum_stock
+        }
+        subject = f'Attention! {stock_item.part.name} is low on stock'
+        html_message = render_to_string('stock/low_stock_notification.html', context)
+        recipients = starred_users.values_list('email', flat=True)
```

Comment:
Need to add a check to ensure that each of the starred users has a valid email

Reply:
**starred_users** is an `allauth.account.models.EmailAddress` QuerySet, so the variable contains the list of available emails of the users who have starred the part in question. The variable name is the cause of confusion I guess. It should have been **starred_users_email**. I shall change it to the latter.